### ISSUE TITLE
request: customizable git commit shortcut

### ISSUE THREAD

**Issue by BastiHz** [rstudio/rstudio#6501]
I have a german keyboard where the key for "M" can also be used to type the lowercase greek letter mu: "µ". Today I updated RStudio to 1.3.921 on Windows 10. Before, typing either AltGr + M or Alt + Ctrl + M produced the letter µ. Now it opens the commit window and I have to use the alt code Alt 230. This is somewhat annyoing because my research is about toxicants in the µg/L concentration range and I use that letter all the time.

It would be great if I could rebind the commit action to some other key combination. Maybe this could be solved in conjunction with #3685?

**Comment by kevinushey** [comment]
What version of RStudio were you using before? Do things work as expected with RStudio v1.2?

RStudio has had this keyboard shortcut bound to <kbd>Ctrl + Alt + M</kbd> for quite some time:

https://github.com/rstudio/rstudio/blame/master/src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml#L814

so it seems more like something may have changed in how keys are handled in this case. Most likely `AltGr` is being interpreted as a "regular" Alt rather than a separate kind of key?

**Comment by kevinushey** [comment]
It looks like this is a regression; I was able to reproduce this as well.

**Comment by BastiHz** [comment]
> What version of RStudio were you using before? Do things work as expected with RStudio v1.2?

I was using 1.2.5033.

**Comment by kevinushey** [comment]
I spent some time digging in, and it looks like this was caused by the upgrade from Qt 5.12.1 to Qt 5.12.6. It looks like we were inadvertently depending on buggy behavior in the Qt 5.12.1 release.

Here's some logging of the JavaScript keydown event we see when one presses <kbd>AltGr + M</kbd>:

## Qt 5.12.1

<img width="689" alt="Screen Shot 2020-03-20 at 2 40 12 PM" src="https://user-images.githubusercontent.com/1976582/77208878-11019880-6aba-11ea-94ae-48593e0442c8.png">

## Qt 5.12.6

<img width="704" alt="Screen Shot 2020-03-20 at 2 43 07 PM" src="https://user-images.githubusercontent.com/1976582/77208904-1fe84b00-6aba-11ea-9c45-e919c9250eed.png">

And for posterity, comparing with Google Chrome:

## Chrome 80.0.3987.149

<img width="683" alt="Screen Shot 2020-03-20 at 2 51 34 PM" src="https://user-images.githubusercontent.com/1976582/77208971-502fe980-6aba-11ea-8563-30f07f3d8981.png">

In other words, there are two issues with Qt here:

1. The older version of Qt did not emit a proper keycode for the <kbd>AltrGr + M</kbd> combination, which is why our shortcut manager didn't see it;

1. Qt doesn't seem to know, or propagate, the difference between the left alt + right alt keys;

1. Qt treats <kbd>AltGr + M</kbd> as though the user had typed <kbd>Ctrl + Alt + M</kbd>.

Unfortunately, Qt WebEngine does not distinguish between the left alt keys and the right alt keys (even though the latest version of Chrome does). In effect, Qt instead converts <kbd>AltGr + M</kbd> to <kbd>Ctrl + Alt + M</kbd>, so they are (almost) indistinguishable.
